csrc/swizzle.cpp

```diff
@@ -70,27 +70,6 @@ std::pair<Val*, Val*> unCyclicShift(Val* x, Val* y, Val* size_x) {
   return {x, mod(sub(add(size_x, y), x), size_x)};
 }
 
-// Scatter swizzle:
-//   Corresponds to the data layout out of ldmatrix intrinsic.
-//   supported dimensions are : 8x4, 16x4, 32x4
-std::pair<Val*, Val*> Scatter(Val* x, Val* y, int64_t size_x) {
-  TORCH_CHECK(
-      size_x == 8 || size_x == 16 || size_x == 32,
-      "Unsupported Scatter swizzle size");
-  Val* size_x_val = IrBuilder::create<Int>(size_x);
-  auto four = IrBuilder::create<Int>(4);
-  return {cpp_div(add(mul(y, size_x_val), x), four), mod(x, four)};
-}
-
-std::pair<Val*, Val*> unScatter(Val* x, Val* y, int64_t size_x) {
-  TORCH_CHECK(
-      size_x == 8 || size_x == 16 || size_x == 32,
-      "Unsupported Scatter swizzle size");
-  Val* size_x_div_4 = IrBuilder::create<Int>(size_x / 4);
-  auto four = IrBuilder::create<Int>(4);
-  return {add(y, mul(mod(x, size_x_div_4), four)), cpp_div(x, size_x_div_4)};
-}
```

Comment:
I think they are abandoned code. `ldmatrix` uses affine swizzle. For example in `Scatter`, the index is
```
(y * size_x_val + x) / 4
x % x
```
This can be un-simplified with reversing `distributeDivisibleDivMod` in expr simplifier as
```
(y * size_x_val + x) / 4
(y * size_x_val + x) % 4
```
Then this is a
```
[size_y_val * size_x_val/4, 4]
merge(0)
[size_y_val * size_x_val]
split(0, size_x_val)
[size_y_val, size_x_val]
```